### PR TITLE
GO-2734 Set global name in new spaces

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-2734/add-cache-for-client-to-resolve-id-name

We should update global names of all participants on:
- new space creation
- registration of new identity
- change of RequestedAnyName on membership status update

### CODE REVIEW

File: core/identity/identity.go [hunk 1]
```diff
@@ -373,17 +382,17 @@ func (s *service) observeIdentitiesLoop() {
 		case <-s.closing:
 			return
 		case <-s.identityForceUpdate:
-			observe()
+			observe(true)
```
Comment (deff7) [inline core/identity/identity.go:385]: it will be called on each RegisterIdentity call

File: core/identity/identity.go [hunk 2]
```diff
@@ -373,17 +382,17 @@ func (s *service) observeIdentitiesLoop() {
...
+			observe(false)
 		}
 	}
 }
 
 const identityRepoDataKind = "profile"
 
 // TODO Maybe we need to use backoff in case of error from coordinator
-func (s *service) observeIdentities(ctx context.Context) error {
+func (s *service) observeIdentities(ctx context.Context, forceUpdate bool) error {
```
Comment (deff7) [inline core/identity/identity.go:395]: not just forceUpdate, but globalNameForceUpdate

File: core/identity/identity.go (hunk 1 above)
Comment (KirillSto) [reply core/identity/identity.go:385]: Yes, it could be a problem?

File: core/identity/identity.go (hunk 2 above)
Comment (KirillSto) [reply core/identity/identity.go]: Done

File: core/identity/identity.go [hunk 3]
```diff
@@ -226,6 +231,13 @@ func (s *service) GetMyProfileDetails() (identity string, metadataKey crypto.Sym
 	return s.myIdentity, s.spaceService.AccountMetadataSymKey(), s.currentProfileDetails
 }
 
+func (s *service) UpdateIdentities() {
```
Comment (deff7) [inline core/identity/identity.go:234]: update global names

**Review by deff7 — APPROVED** [review]

**Review by AnthonyAkentiev — APPROVED** [review]

File: core/identity/identity.go (hunk 3 above)
Comment (KirillSto) [reply core/identity/identity.go]: Done

**Pull request merged**